python_modules/dagster/dagster/_core/events/__init__.py

```diff
@@ -1504,14 +1511,31 @@ def __new__(
 class AssetMaterializationPlannedData(
     NamedTuple(
         "_AssetMaterializationPlannedData",
-        [("asset_key", AssetKey), ("partition", Optional[str])],
+        [
+            ("asset_key", AssetKey),
+            ("partition", Optional[str]),
+            ("partitions_subset", Optional["PartitionsSubset"]),
```

Comment:
This needs to be serializable, and not all PartitionsSubsets are serializable, right? Should we check for this in the constructor?

Reply:
Ah good call, right now we convert the subset to a serializable subset at the callsite, but probably safer to have a check here too